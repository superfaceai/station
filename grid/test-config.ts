import { NockConfig } from '@superfaceai/testing';

export const nockConfig: NockConfig = {
  testInstance: expect,
};
