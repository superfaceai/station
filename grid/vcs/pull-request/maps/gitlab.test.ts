import { SuperfaceTest } from '@superfaceai/testing';

import { buildSuperfaceTest } from '../../../test-config';

describe(`vcs/pull-request/gitlab}`, () => {
  let superface: SuperfaceTest;

  beforeEach(() => {
    superface = buildSuperfaceTest({
      profile: 'vcs/pull-request',
      provider: 'gitlab',
    });
  });

  describe('PullRequest', () => {
    it('should perform successfully', async () => {
      await expect(
        superface.run({
          useCase: 'PullRequest',
          input: {
            owner: 'Jakub-Vacek',
            repo: 'empty-test',
            identifier: 1,
          },
        })
      ).resolves.toMatchSnapshot();
    });

    it('should map error', async () => {
      await expect(
        superface.run({
          useCase: 'PullRequest',
          input: {
            owner: 'Jakub-Vacek',
            repo: 'empty-test',
            identifier: 9999,
          },
        })
      ).resolves.toMatchSnapshot();
    });
  });
});
