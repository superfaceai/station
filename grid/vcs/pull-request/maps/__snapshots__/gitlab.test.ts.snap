// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vcs/pull-request/gitlab} PullRequest should map error 1`] = `
Err {
  "error": "MappedHTTPError: Expected HTTP error
Properties: {
  \\"message\\": \\"404 Project Not Found\\",
  \\"statusCode\\": 404
}
AST Path: definitions[0].statements[2].responseHandlers[0].statements[1]
Original Map Location: Line 20, column 7",
}
`;

exports[`vcs/pull-request/gitlab} PullRequest should perform successfully 1`] = `
Err {
  "error": "MappedHTTPError: Expected HTTP error
Properties: {
  \\"message\\": \\"404 Project Not Found\\",
  \\"statusCode\\": 404
}
AST Path: definitions[0].statements[2].responseHandlers[0].statements[1]
Original Map Location: Line 20, column 7",
}
`;
